opentelemetry/proto/profiles/v1/alternatives/arrays/arrays.proto

```diff
@@ -0,0 +1,169 @@
+  bytes trace_id = 1;
+  // A unique identifier for the linked span. The ID is an 8-byte array.
+  bytes span_id = 2;
+}
+
+// AttributeSet represents a set of attributes. It is referenced from Samples
+// and not embedded because it is common for multiple samples to have the same
+// set of attributes
+message AttributeSet {
+  repeated opentelemetry.proto.common.v1.KeyValue attributes = 1;
+  uint32 dropped_attributes_count = 2;
+}
+
+message Stacktrace {
+  repeated uint32 location_indices = 1;
+}
+
+// message Sample {
+//   uint32 stacktrace_index = 1;
+//   repeated uint32 link_indices = 2;
+//   repeated uint32 attribute_set_indices = 3;
+//   repeated int64 values = 4;
+
+//   // optional
+//   fixed64 timestamp_unix_nano = 5;
+// }
+
+message SampleType {
+  // borrowed from metrics proto
+  enum AggregationTemporality {
+    AGGREGATION_TEMPORALITY_UNSPECIFIED = 0;
+    AGGREGATION_TEMPORALITY_DELTA = 1;
+    AGGREGATION_TEMPORALITY_CUMULATIVE = 2;
+  }
+
+  AggregationTemporality aggregation_temporality = 1;
+  uint64 sample_size = 2;
+
+  // CPU / memory /etc
+  uint32 type_index = 3; // Index into string table.
+  uint32 unit_index = 4; // Index into string table.
+}
+
+message ProfileType {
+  repeated uint32 stacktrace_indices = 10;
+  repeated uint32 link_indices = 11;
+  repeated uint32 attribute_set_indices = 12;
+  repeated int64 values = 13;
+  repeated uint64 timestamps = 14;
+}
+
+message Profile {
+  repeated SampleType sample_types = 1;
+  repeated Stacktrace stacktraces = 2;
+  repeated Mapping mappings = 4;
+  repeated Location locations = 5;
+  repeated Function functions = 6;
+  repeated Link links = 7;
+  repeated AttributeSet attribute_sets = 8;
+
+  // 0-th element must be an empty string ("")
+  repeated string string_table = 9;
+
+  repeated ProfileType profile_types = 10;
+
+  // stacktrace_index
+  // link_indices
+  // attribute_set_indices
+  // values
+  // timestamp_unix_nano
+}
+
+
+// borrowed from pprof proto
+message Mapping {
+  // Address at which the binary (or DLL) is loaded into memory.
+  uint64 memory_start = 1;
+  // The limit of the address range occupied by this mapping.
+  uint64 memory_limit = 2;
+  // Offset in the binary that corresponds to the first mapped address.
+  uint64 file_offset = 3;
+  // The object this entry is loaded from.  This can be a filename on
+  // disk for the main binary and shared libraries, or virtual
+  // abstractions like "[vdso]".
+  uint32 filename_index = 4;  // Index into string table
+  // A string that uniquely identifies a particular program version
+  // with high probability. E.g., for binaries generated by GNU tools,
+  // it could be the contents of the .note.gnu.build-id field.
+  uint32 build_id_index = 5;  // Index into string table
+
+  enum SymbolicInfo {
+    SYMBOLIC_INFO_UNSPECIFIED = 0;
+    SYMBOLIC_INFO_FULL = 1;
+    SYMBOLIC_INFO_FUNCTIONS_ONLY = 2;
+    SYMBOLIC_INFO_NO_INLINE_FRAMES = 3;
+  }
+
+  SymbolicInfo symbolic_info = 6;
```

Comment:
Instead of `SymbolicInfo` we maybe should use `repeated Label` here. Similar to the original discussion https://github.com/open-telemetry/opentelemetry-proto/pull/488#discussion_r1246417203